Library/Homebrew/test/gpg_spec.rb

```diff
@@ -13,7 +13,12 @@
         shutup do
           subject.create_test_key(dir)
         end
-        expect(dir/".gnupg/secring.gpg").to exist
+
+        begin
+          expect(dir/".gnupg/pubring.kbx").to be_file
+        rescue RSpec::Expectations::ExpectationNotMetError
+          expect(dir/".gnupg/secring.gpg").to be_file
+        end
```

Comment:
Ok, the core issue here is that `subject.create_test_key(dir)` can potentially invoke either gnupg 2.1.x or more rarely gnupg 2.0.x.

This is because `Gpg::find_gpg` will set `GPG_EXECUTABLE` to whichever version it finds first in the path. In turn, `Gpg::create_test_key` invokes this executable to generate the keystore. This is a problem, because 2.1.x will create a `pubring.kbx` file but not a `secring.gpg` file while the reverse is true for 2.0.x. 

Most likely it will find gnupg 2.1.x, since that is now the default formula, and the first expectation will pass. However, if gnupg 2.0.x is found first, the test will still be run and it is guaranteed the first expectation will fail. Therefore, that failure needs to be handled and the expectation designed for 2.0.x evaluated instead.

Hopefully that describes what I am trying to do with that code sufficiently.